### PR TITLE
Add no cookie option

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Closes https://github.com/PostHog/posthog/issues/1571

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.4.4 - 2020-08-26
```
Comment (macobo) [inline CHANGELOG.md:1]: This seems out of date

File: src/posthog-persistence.js [hunk 2]
```diff
@@ -59,6 +59,9 @@ var PostHogPersistence = function (config) {
     } else {
         this.storage = _.cookie
     }
+    if (storage_type === 'disabled') {
```
Comment (macobo) [inline src/posthog-persistence.js:62]: Wouldn't it make sense to make this into:

```
    if (storage_type === 'localStorage' && _.localStorage.is_supported()) {
        this.storage = _.localStorage
    } else if (storage_type === 'disabled') {
        this.storage = _.storageDisabled
    } else {
        this.storage = _.cookie
    }
```

**Review by macobo — COMMENTED** [review]
Both autocapture.js and posthog-featureflags.js have a check like the following:

```
                instance.persistence &&
                    instance.persistence.register({ $active_feature_flags: response['featureFlags'] })
```

Does the `instance.persistence &&` check serve any purpose anymore? If not, perhaps worth killing it?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "posthog-js",
-    "version": "1.4.3",
+    "version": "1.4.4",
```
Comment (mariusandra) [inline package.json:3]: I think we should bump versions outside PRs... This is hard to merge now and should be rebased

**Review by mariusandra — CHANGES_REQUESTED** [review]
@timgl great for this feature, as some users really don't want to show cookie banners and this helps with that. 

However,
- Missing tests
- Conflicts with package.json and changelog
- Nit from @macobo regarding an if statement

:)

**Comment by macobo** [comment]
Since this has been sitting in limbo, I took this over, refactored a bit and fixed a few issues.

File: src/storage.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+            if (is_secure) {
+                secure = '; secure'
+            }
+
+            document.cookie = name + '=' + encodeURIComponent(value) + expires + '; path=/' + cdomain + secure
+        } catch (err) {
+            return
+        }
+    },
```
Comment (mariusandra) [inline src/storage.js:61]: I couldn't find anything using this function. Dead code?

File: src/storage.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    remove: function (name) {
+        try {
+            window.localStorage.removeItem(name)
+        } catch (err) {
+            localStore.error(err)
+        }
+    },
+}
+
+const disabledStorage = {}
```
Comment (mariusandra) [inline src/storage.js:169]: nit: could also call this `memoryStorage` after the rename

File: src/posthog-persistence.js [hunk 6]
```diff
@@ -140,11 +143,11 @@ PostHogPersistence.prototype.upgrade = function (config) {
...
 
-    if (this.storage === _.localStorage) {
-        old_cookie = _.cookie.parse(this.name)
+    if (this.storage !== cookieStore) {
+        old_cookie = cookieStore.parse(this.name)
 
-        _.cookie.remove(this.name)
-        _.cookie.remove(this.name, true)
+        cookieStore.remove(this.name)
+        cookieStore.remove(this.name, true)
```
Comment (mariusandra) [inline src/posthog-persistence.js:150]: nit: this feels kind of dirty if moving from memory -> localStorage for example, but I think all is good and better be safe with the cookies...

**Review by mariusandra — APPROVED** [review]
Small nits and a bit of dead code, otherwise looking 💯 . Fix if you wish, merge otherwise :)

File: src/storage.js (hunk 4 above)
Comment (macobo) [reply src/storage.js]: Seems so, great catch!

**Pull request merged**